File: week_03/src/Question_22.js
// Author : Zaid Nayeb
// Date   : 2025/11/26
/*
22. Collatz Sequence (Hailstone Numbers) Description: 
Take a number n. 
•	If n is even, divide it by 2. 
•	If n is odd, multiply it by 3 and add 1. 
•	Repeat until n = 1. 
•	Count the number of steps to reach 1. 
Example: 
Input: 
6 
Output: 
6 → 3 → 10 → 5 → 16 → 8 → 4 → 2 → 1 (Steps: 8) 


*/
console.log(coolatz(6));

function coolatz(num) {
    let counter = 0;
    let output = num + '';

    while (num != 1) {
        num % 2 === 0 ? num /= 2 : num = num * 3 + 1;
        output += " -> " + num;
        counter++;
    }
    return output + " (Steps : " + counter + ")";

}

module.exports = coolatz;

File: week_02/src/main/java/Question22.java
/*
Author = Zaid Nayeb
Date   = 18/11/2025

22.	Print even numbers in a range
Description: Input start and end values, print all even numbers between them.
Example:
 Input: 1, 10
Output: 2, 4, 6, 8, 10

*/
public class Question22 {
    public static String printEven(int a, int b){
        StringBuilder numbers = new StringBuilder();
        for (int i = a; i <= b; i++){
            if ( i % 2 == 0 ){
                numbers.append(i);
                if (i < b ){
                    numbers.append(", ");
                }
            }
        }
        return numbers.toString();
    }
}
